src/main/java/org/terasology/climateConditions/HypothermiaSystem.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.climateConditions;
+
+import org.terasology.alterationEffects.speed.*;
+import org.terasology.biomesAPI.OnBiomeChangedEvent;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.entity.lifecycleEvents.OnAddedComponent;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.characters.AffectJumpForceEvent;
+import org.terasology.logic.characters.CharacterMovementComponent;
+import org.terasology.logic.characters.GetMaxSpeedEvent;
+import org.terasology.logic.characters.MovementMode;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.logic.players.PlayerCharacterComponent;
+import org.terasology.physics.events.MovedEvent;
+
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class HypothermiaSystem extends BaseComponentSystem {
+    private float thresholdHeight = 60f;
+    private float WalkSpeedMultiplier = 0.6f;
+    private float JumpSpeedMultiplier = 0.7f;
+
+    @ReceiveEvent(components = {PlayerCharacterComponent.class, CharacterMovementComponent.class})
+    public void observeDangerZone(MovedEvent event, EntityRef player, LocationComponent location, CharacterMovementComponent movement) {
+        //TODO: react on OnBiomeChangedEvent to handle the danger zone
+        float height = event.getPosition().getY();
+        float lastHeight = height - event.getDelta().getY();
+        if (height > thresholdHeight && lastHeight <= thresholdHeight) {
+            player.addOrSaveComponent(new HypothermiaComponent());
+        }
+        if (height < thresholdHeight && lastHeight >= thresholdHeight) {
+            if(player.hasComponent(HypothermiaComponent.class)){
+                player.removeComponent(HypothermiaComponent.class);
+            }
+        }
+    }
+
+    @ReceiveEvent
+    public void modifySpeed(GetMaxSpeedEvent event, EntityRef player) {
+        // If the entity is walking and they have a walk speed effect, boost their walking speed by the multiplier.
+        if (event.getMovementMode() == MovementMode.WALKING && player.hasComponent(HypothermiaComponent.class)) {
```

Comment:
Was limiting the effect to affect only the movement mode "walking" a deliberate decision? If so, I'd like to learn why :)

Reply:
Well not really, but Walking and Running both come under Walking and I haven't seen any water bodies in Snow yet so... didn't add any alteration for swimming yet... And when and if I do add alteration for swimming.. I'd like to modify it a bit more... The water + snow should have a stronger effect on the body of the player thing...

Reply:
Sounds like that you can want to add another module with cloth and "cold" resistance :)
And wet factor....
Ahh, just another random idea  :P

Reply:
Don't know about another module 😅 but would definitely want the wet factor thing in extreme cold to have a magnified effect on the player..

Reply:
I see :+1: 
I'm still not entirely sure hypothermia should _only_ be related to the biome being a snow biome, but we can come back to that later, so that's fine for now :slightly_smiling_face:

Reply:
@DarkWeird bringin clothing into the equation is exactly what @ktksan put into their proposal 😉 

Another approach on this could be to let the effect apply to any movement mode, and then refine it further from top-to-bottom, so to say. But as @jdrueckert said, we can come back to this later 🙃